packages/remix-dev/vite/plugin.ts

```diff
@@ -477,6 +479,28 @@ export const remixVitePlugin: RemixVitePlugin = (options = {}) => {
           }),
         };
       },
+      // after ssr asset generation, move them to client assets directory
+      // https://rollupjs.org/plugin-development/#writebundle
```

Comment:
As indicated in https://github.com/vitejs/vite/pull/11430#issuecomment-1453462820, one thing this approach could be missing is that vite's client `manifest.json` won't include entries from ssr assets.
I saw vite manifest is used in `createBuildManifest/resolveBuildAssetPaths`, but ssr assets are not something hard-coded on client routes, so I don't think it's useful for the purpose of per-route resource pre-fetching etc...